iOS/Flows/Conversation/Input/InputHandlerCoordinator.swift

```diff
@@ -0,0 +1,179 @@
+//
+//  InputHandlerCoordinator.swift
+//  Jibber
+//
+//  Created by Benji Dodgson on 3/23/22.
+//  Copyright © 2022 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+import PhotosUI
+import Photos
+import Localization
+
+protocol SwipeableInputControllerHandler where Self: ViewController {
+    var swipeableVC: SwipeableInputAccessoryViewController { get }
+    func updateUI(for state: ConversationUIState, forceLayout: Bool)
+}
+
+typealias InputHandlerViewContoller = SwipeableInputControllerHandler & ViewController
+
+class InputHandlerCoordinator<Result>: PresentableCoordinator<Result>,
+                                       ActiveConversationable,
+                                       PHPickerViewControllerDelegate,
+                                       UIImagePickerControllerDelegate,
+                                       UINavigationControllerDelegate {
+    
+    lazy var pickerVC: PHPickerViewController = {
+        var filter = PHPickerFilter.any(of: [.images])
+        var config = PHPickerConfiguration(photoLibrary: .shared())
+        config.filter = filter
+        config.selectionLimit = 1
+        let vc = PHPickerViewController(configuration: config)
+        vc.delegate = self
+        return vc
+    }()
+    
+    lazy var captureVC: UIImagePickerController = {
+        let vc = UIImagePickerController()
+        vc.sourceType = .camera
+        vc.allowsEditing = true
+        vc.delegate = self
+        return vc
+    }()
+    
+    let inputHandlerViewController: InputHandlerViewContoller
+    
+    init(with viewContoller: InputHandlerViewContoller,
+         router: Router,
+         deepLink: DeepLinkable?) {
+        
+        self.inputHandlerViewController = viewContoller
+        super.init(router: router, deepLink: deepLink)
+    }
+    
+    override func toPresentable() -> PresentableCoordinator<Void>.DismissableVC {
+        return self.inputHandlerViewController
+    }
+    
+    override func start() {
+        super.start()
+        
+        self.inputHandlerViewController.swipeableVC.swipeInputView.addView.didSelect { [unowned self] in
+            self.presentAttachements()
+        }
+    }
+    
+    func presentAttachements() {
+        let coordinator = AttachmentsCoordinator(router: self.router, deepLink: self.deepLink)
+        
+        self.present(coordinator) { [unowned self] result in
+            self.handle(attachmentOption: result)
+        }
+    }
+    
+    func present<ChildResult>(_ coordinator: PresentableCoordinator<ChildResult>,
```

Comment:
Both share the need for this presentation logic. Probably fixed a lot of bugs by adding this.